src/pykeen/pipeline.py

```diff
@@ -709,6 +754,12 @@ def pipeline(  # noqa: C901
     validation: Union[None, TriplesFactory, str] = None,
     evaluation_entity_whitelist: Optional[Collection[str]] = None,
     evaluation_relation_whitelist: Optional[Collection[str]] = None,
+    # Interaction
+    interaction: Union[str, Type[Interaction], Interaction],
+    interaction_kwargs: Optional[Mapping[str, Any]] = None,
+    # Representations
+    entity_representations: OneOrMany[Union[None, RepresentationModule, EmbeddingSpecification]],
+    relation_representations: OneOrMany[Union[None, RepresentationModule, EmbeddingSpecification]],
```

Comment:
@mberr 
Can you update the docstring accordingly?
Currently none of those examples would work.

Reply:
Yes, that is due to being only partially implemented. If we want to have the pipeline changes in this PR, we need to finish that. Otherwise, f30b73b goes into a new PR, and 
c1b6194 is the HEAD for this PR.

Reply:
I vote for it being done in this PR, since those changes are directly related to the overall model API changes of this PR.

Reply:
I moved the pipeline changes to a new PR #163, which targets this branch. Thereby, we can keep them separated and decide which parts we want to have in master later without blocking progress.